### PR TITLE
Add started_at and errored_at columns to Distribution

### PR DESCRIPTION
**Pull request by pkarman**
connects #11713 

A bunch of misc clean up in prep for understanding how Distributions work.

Moves some files, refactors a large method into into its own class, fixes a small js bug.

Adds new columns for the `Distribution` model so we can get a better sense of job timings.

*Schema Changes Only*

* [x] Column comments updated
* [x] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [ ] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: client/app/components/Pagination.jsx [hunk 1]
```diff
@@ -50,11 +50,11 @@ class Pagination extends React.PureComponent {
 
     // If there are no pages, there is no data, so the range should be 0-0.
     // Otherwise, the beginning of the range is the previous amount of cases + 1
-    const beginningCaseNumber = (currentPage * pageSize) - pageSize + 1;
+    const beginningCaseNumber = totalCases === 0 ? 0 : ((currentPage * pageSize) - pageSize + 1);
```
Comment (pkarman) [inline client/app/components/Pagination.jsx]: this fixes a pagination bug where it was displaying `1-0 of 0 total`

File: app/controllers/distributions_controller.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 class DistributionsController < ApplicationController
+  include RunAsyncable
```
Comment (pkarman) [inline app/controllers/distributions_controller.rb]: just uses the concern -- no functional change

File: app/jobs/start_distribution_job.rb [hunk 3]
```diff
@@ -17,6 +17,6 @@ def perform(distribution, user = nil)
   def handle_error(error)
     Rails.logger.info "StartDistributionJob failed: #{error.message}"
     Rails.logger.info error.backtrace.join("\n")
-    Raven.capture_exception(error)
```
Comment (pkarman) [inline app/jobs/start_distribution_job.rb]: new job style

File: app/models/distributed_case.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ class DistributedCase < ApplicationRecord
   belongs_to :distribution
   belongs_to :task
 
-  validates :distribution, :case_id, :docket, :ready_at, presence: true
+  validates :case_id, :distribution, :docket, :ready_at, presence: true
```
Comment (pkarman) [inline app/models/distributed_case.rb]: just some alphabetizing

File: app/models/distribution.rb [hunk 5]
```diff
@@ -33,9 +33,14 @@ def distribute!
       return unless valid?(context: :create)
     end
 
+    # always set the timestamp, ok if the status rolls back in the transaction below.
+    update!(started_at: Time.zone.now)
```
Comment (pkarman) [inline app/models/distribution.rb]: using the new columns

File: app/models/vacols/case.rb [hunk 6]
```diff
@@ -253,18 +253,71 @@ class InvalidLocationError < StandardError; end
...
   }
 
-  # These scopes query VACOLS and cannot be covered by automated tests.
-  # :nocov:
-  def self.remands_ready_for_claims_establishment
-    VACOLS::Case.joins(:folder, :correspondent)
-      .where(WHERE_PAPERLESS_REMAND_LOC97)
-      .order("BFDDEC ASC")
-  end
+  class << self
```
Comment (pkarman) [inline app/models/vacols/case.rb]: reorganized class methods per #11778 and refactored `batch_update_vacols_location` into its own class.

File: app/queries/vacols_location_batch_updater.rb [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class VacolsLocationBatchUpdater
```
Comment (pkarman) [inline app/queries/vacols_location_batch_updater.rb]: refactor from VACOLS::Case

File: spec/models/distribution_spec.rb [hunk 8]
```diff
@@ -189,12 +192,37 @@
       expect(subject.distributed_cases.where(docket: Constants.AMA_DOCKETS.evidence_submission).count).to eq(2)
     end
 
+    context "when a nonpriority legacy case re-distribtution is attempted" do
```
Comment (pkarman) [inline spec/models/distribution_spec.rb]: this is the meaningful test. it validates current behavior (which we might change later).

File: app/models/vacols/case.rb (hunk 6 above)
Comment (lomky) [reply app/models/vacols/case.rb]: Other than this, was the functionality changed at all?

Comment (pkarman) [reply app/models/vacols/case.rb]: nope, same functionality. just a refactor to make a new Query class.

**Pull request merged**